skeleton/README.md

```diff
@@ -0,0 +1,152 @@
+
+This is an empty reference project which shows how to set up CI with
+Hydra and Buildkite, using the `iohk-nix` library.
+
+## How to set up a project
+
+1. Copy this directory to a new location and then:
+   ```
+   git init
+   git commit --allow-empty -m "Initial commit"
+   git add .
+   git commit -m "Copy iohk-skeleton"
+   ```
+
+2. Run `./nix/update-iohk-nix.sh` to update the iohk-nix pin.
+
+3. Run `./nix/regenerate.sh` to regenerate Nix expressions (nothing
+   will change, but run it to make sure).
+
+4. Find all instances of `iohk-skeleton` or `iohkSkeleton` and replace
+   them with your project name. Find all instances of `TODO` and
+   address them. You can use the following tool to do that:
+
+   ```
+   nix-shell --run "lentil -w iohkSkeleton -w iohk-skeleton -T fixme ."
+   ```
+
+5. Follow the subsequent instructions in this document. The referenced
+   config files should also have example code and comments with
+   instructions.
+
+6. When finished, replace this file with your own, and delete
+   redundant comments from the config files.
+
+7. Run `./nix/regenerate.sh` to regenerate Nix expressions whenever
+   `*.cabal` or `stack.yaml` have changed. Add and remove the
+   generated files from git.
+
+8. If something in this project needs to be improved, please
+   [open a PR](https://github.com/input-output-hk/iohk-nix/pulls).
+   If you have any other problems, ask in
+   [#devops-crossteam-ci](https://app.slack.com/client/T0N639Z4N/CAP8NM7N0).
+
+
+## How to push to GitHub
+
+1. Create a new repo in GitHub. It's generally safest to create a
+   private repo first, then later get an organization admin to change
+   it to public.
+
+   However, note that Buildkite and Hydra won't work until the repos
+   are made public.
+
+2. In the GitHub repo _Collaborators & teams_ section, grant `Write` access to the
+   [cardano-sl](https://github.com/orgs/input-output-hk/teams/cardano-sl)
+   team, and `Admin` access to the
+   [cardano-sl-admin](https://github.com/orgs/input-output-hk/teams/cardano-sl-admin)
+   team.
+
+
+## How to set up Buildkite
+
+Use Buildkite for CI jobs which can't run in a pure sandboxed build
+environment. Examples of these would be tests which need Internet
+access, benchmarks, stack builds, or jobs which update a website.
+
+1. Create a [New Pipeline](https://buildkite.com/organizations/input-output-hk/pipelines/new)
+   in Buildkite for the project repo.
+
+2. Add a single step from the template called "Read steps from repository".
+
+   In this step, set the _Agent Targeting Rules_ to `system=x86_64-linux`
+
+3. Create the pipeline. Follow the instructions to set up the GitHub
+   webhook for Buildkite.
+
+4. Edit the pipeline settings. Enable the options:
+    - Skip Intermediate Builds
+    - Cancel Intermediate Builds
+
+5. Grab the Build Badge link and put it in this file.
+
+6. The pipeline definition is in [`.buildkite/pipeline.yml`](./.buildkite/pipeline.yml).
```

Comment:
Is there a way we can test the `buildkite` configuration (and associated `nix` files) locally?